src/parser/jobs/sch/modules/Interrupts.js

```diff
@@ -0,0 +1,112 @@
+import React from 'react'
+import {Table, Button} from 'semantic-ui-react'
+import {Trans} from '@lingui/react'
+import {t} from '@lingui/macro'
+
+import Module from 'parser/core/Module'
+import {getDataBy} from 'data'
+import ACTIONS from 'data/ACTIONS'
+import {ActionLink} from 'components/ui/DbLink'
+import {TieredSuggestion, SEVERITY} from 'parser/core/modules/Suggestions'
+
+const TIMELINE_UPPER_MOD = 30000
+
+const INTERRUPT_SEVERITY = {
+	2: SEVERITY.MEDIUM,
+	5: SEVERITY.MAJOR,
+}
+
+export default class Interrupts extends Module {
+	static handle = 'interrupts'
+	static title = t('sch.interrupts.title')`Interrupted Casts`
+
+	static dependencies = [
+		'timeline',
+		'gcd',
+		'suggestions',
+	]
+
+	_currentCast = null
```

Comment:
nit; pref. `undefined` over `null`

It's _stupid_ that JS even has both, but TS makes it _much_ nicer to deal with `undefined` than it does `null` (`foo?: bar` vs `foo: bar | null`).